R/plot.R

```diff
@@ -780,3 +780,519 @@ make_waterfall_plot <- function(dt_plot,
+#'   MSEv_figure_wo_error_bars +
+#'     ggplot2::geom_text(ggplot2::aes(label = sprintf(paste("%.", sprintf("%d", bar_text_n_decimals), "f", sep = ""),
+#'                                                     round(MSEv, bar_text_n_decimals))),
+#'                        vjust = 1.75,
+#'                        hjust = NA,
+#'                        color = "black",
+#'                        position = ggplot2::position_dodge(0.9),
+#'                        size = 5)
+#'
+#'   # Rotate the plot
+#'   MSEv_figure +
+#'     ggplot2::scale_x_discrete(limits = rev(levels(MSEv_figure$data$Method))) +
+#'     ggplot2::coord_flip()
+#'
+#'   # All of these can be combined
+#'   MSEv_figure_wo_error_bars +
+#'     ggplot2::scale_x_discrete(limits = rev(levels(MSEv_figure$data$Method))) +
+#'     ggplot2::coord_flip() +
+#'     ggplot2::scale_fill_discrete() + #' Default ggplot2 palette
+#'     ggplot2::theme_minimal() + #' This must be set before the other theme call
+#'     ggplot2::theme(plot.title = ggplot2::element_text(size = 10),
+#'                    legend.position = "bottom") +
+#'     ggplot2::guides(fill = ggplot2::guide_legend(nrow = 1, ncol = 6)) +
+#'     ggplot2::geom_text(ggplot2::aes(label = sprintf(paste("%.", sprintf("%d", bar_text_n_decimals), "f", sep = ""),
+#'                                                     round(MSEv, bar_text_n_decimals))),
+#'                        vjust = NA,
+#'                        hjust = 1.15,
+#'                        color = "black",
+#'                        position = ggplot2::position_dodge(0.9),
+#'                        size = 5)
+#'
+#'
+#'   # Can also create plots where we look at the MSEv criterion averaged only over the combinations or observations.
+#'   # Note that we can also alter the design of these plots as we did above.
+#'   MSEv_figures = make_MSEv_eval_crit_plots(explanation_list_named,
+#'                                            make_MSEv_comb_and_explicand = TRUE,
+#'                                            N_sd_error_bars = 0)
+#'   MSEv_figures$MSEv_bar
+#'   MSEv_figures$MSEv_combination_bar
+#'   MSEv_figures$MSEv_explicand_bar
+#'
+#'   # When there are many combinations or observations, then it can be easier to look at line plots
+#'   MSEv_figures$MSEv_combination_line_point
+#'   MSEv_figures$MSEv_explicand_line_point
+#'
+#'   # We can specify which test observations or combinations to plot
+#'   make_MSEv_eval_crit_plots(explanation_list_named,
+#'                             make_MSEv_comb_and_explicand = TRUE,
+#'                             index_x_explain = c(1, 3:4, 6),
+#'                             N_sd_error_bars = 1)$MSEv_explicand_bar
+#'   make_MSEv_eval_crit_plots(explanation_list_named,
+#'                             make_MSEv_comb_and_explicand = TRUE,
+#'                             id_combination = c(3, 4, 9, 13:15),
+#'                             N_sd_error_bars = 1)$MSEv_combination_bar
+#' }
```

Comment:
I think this is too much and make the help files very long. And even if the plotting is done outside of shapr it takes a bit of time. I think we should reduce it significantly. We can instead put this in a script or something and put it on github.

Reply:
Shorted the examples.  Put the other examples into the `example_plot_MSEv.R` file inside the `shapr/inst/scripts` folder.